src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingServiceRemotingClient.cs

```diff
@@ -0,0 +1,177 @@
+
+        /// <summary>
+        /// Gets or Sets the name of the listener in the replica or instance to which the client is connected to.
+        /// </summary>
+        public string ListenerName { get => this.InnerClient.ListenerName; set => this.InnerClient.ListenerName = value; }
+
+        /// <summary>
+        /// Gets or Sets the service endpoint to which the client is connected to.
+        /// </summary>
+        public ResolvedServiceEndpoint Endpoint { get => this.InnerClient.Endpoint; set => this.InnerClient.Endpoint = value; }
+
+        /// <summary>
+        /// Gets the inner client which this client wraps.
+        /// </summary>
+        public IServiceRemotingClient InnerClient { get; private set; }
+
+        /// <summary>
+        /// Sends a message to the service and gets a response back. The correlation id and context are sent along with
+        /// the message as message headers.
+        /// </summary>
+        /// <param name="messageHeaders">Message headers.</param>
+        /// <param name="requestBody">Message body.</param>
+        /// <returns>Response body.</returns>
+        public Task<byte[]> RequestResponseAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            return SendAndTrackRequestAsync(messageHeaders, requestBody, () => this.InnerClient.RequestResponseAsync(messageHeaders, requestBody));
+        }
+
+        /// <summary>
+        /// Sends a one-way message to the service. The correlation id and context are sent along with the message
+        /// as message headers.
+        /// </summary>
+        /// <param name="messageHeaders">Message headers.</param>
+        /// <param name="requestBody">Message body.</param>
+        public void SendOneWay(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            SendAndTrackRequestAsync(messageHeaders, requestBody, () =>
+            {
+                this.InnerClient.SendOneWay(messageHeaders, requestBody);
+                return Task.FromResult<byte[]>(null);
+            }).Forget();
+        }
+
+        private async Task<byte[]> SendAndTrackRequestAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody, Func<Task<byte[]>> doSendRequest)
+        {
+            DependencyTelemetry dt = new DependencyTelemetry();
+            dt.Name = ServiceRemotingLoggingStrings.OutboundRequestActivityName;
+
+            // Determine if there is currently an activity. If there is, initializes the dependency telemetry with the activity information,
+            // which helps set the right context for the dependency telemetry
+            Activity currentActivity = Activity.Current;
+            if (currentActivity != null)
+            {
+                dt.Id = currentActivity.Id;
+                dt.Context.Operation.Id = currentActivity.RootId;
+                dt.Context.Operation.ParentId = currentActivity.ParentId;
+            }
+
```

Comment:
is there anything interesting about the telemetry we could report?
e.g. for Http dependencies we report Target (target service name), Data (full URL).

We also need to report dependency type (e.g. SF)